components/match2/wikis/counterstrike/match_summary.lua

```diff
@@ -0,0 +1,486 @@
+
+	if match.comment then
+		local comment = MatchSummary.Comment():content(match.comment)
+		comment.root:css('display', 'block'):css('text-align', 'center')
+		matchSummary:comment(comment)
+	end
+
+	local vods = {}
+	for index, game in ipairs(match.games) do
+		if game.vod then
+			vods[index] = game.vod
+		end
+	end
+
+	if not Table.isEmpty(vods) or not Table.isEmpty(match.links) or not Logic.isEmpty(match.vod) then
+		matchSummary:footer(CustomMatchSummary._createFooter(match, vods))
+	end
+
+	return matchSummary:create()
+end
+
+function CustomMatchSummary._createHeader(match)
+	local header = MatchSummary.Header()
+
+	header:leftOpponent(header:createOpponent(match.opponents[1], 'left'))
+	      :leftScore(header:createScore(match.opponents[1]))
+	      :rightScore(header:createScore(match.opponents[2]))
+	      :rightOpponent(header:createOpponent(match.opponents[2], 'right'))
+
+	return header
+end
+
+function CustomMatchSummary._createBody(match)
+	local body = MatchSummary.Body()
+
+	if match.dateIsExact or (match.date ~= EPOCH_TIME_EXTENDED and match.date ~= EPOCH_TIME) then
+		-- dateIsExact means we have both date and time. Show countdown
+		-- if match is not epoch=0, we have a date, so display the date
+		body:addRow(MatchSummary.Row():addElement(
+			DisplayHelper.MatchCountdownBlock(match)
+		))
+	end
+
+	-- Iterate each map
+	for _, game in ipairs(match.games) do
+		if game.map then
+			body:addRow(CustomMatchSummary._createMap(game))
+		end
+	end
+
+	-- Add the Map Vetoes
+	if match.extradata.mapveto then
+		local vetoData = match.extradata.mapveto
+		if vetoData then
+			local mapVeto = MapVeto()
+
+			for _,vetoRound in ipairs(vetoData) do
+				if vetoRound.vetostart then
+					mapVeto:vetoStart(tonumber(vetoRound.vetostart))
+				end
+				if vetoRound.type == 'decider' then
+					mapVeto:addDecider(CustomMatchSummary._createMapLink(vetoRound.decider, match.game))
+				else
+					mapVeto:addRound(vetoRound.type,
+										CustomMatchSummary._createMapLink(vetoRound.team1, match.game),
+										CustomMatchSummary._createMapLink(vetoRound.team2, match.game))
+				end
+			end
+
+			body:addRow(mapVeto)
+		end
+	end
+
+	-- Match Status (postponed/ cancel(l)ed)
+	if match.extradata.status then
+		local matchStatus = MatchStatus()
+		local lang = mw.getContentLanguage()
+		matchStatus:content('\'\'\'Match ' .. lang:ucfirst(match.extradata.status) .. '\'\'\'')
+		body:addRow(matchStatus)
+	end
+
+	return body
+end
+
+function CustomMatchSummary._createFooter(match, vods)
+	local footer = MatchSummary.Footer()
+
+	local separator = '<b>·</b>'
+
+	local function createFooterLink(icon, url, label, index)
+		local _icon = icon
+		if icon == 'stats' then
+			_icon = index ~= 0 and 'Match Info Stats' .. index .. '.png' or 'Match Info Stats.png'
+		end
+		if index > 0 then
+			label = label .. ' for Game ' .. index
+		end
+
+		return '[[FILE:' .. _icon .. '|link=' .. url .. '|15px|' .. label .. '|alt=' .. url .. ']]'
```

Comment:
i don't like `_icon`
maybe a better name?
(another option could be to not use it a t all and just overwrite `icon` in line 306)

Reply:
Overwrite icon is the better option, as we also overwrite label